polygeist/include/mlir/Dialect/Polygeist/Utils/TransformUtils.h

```diff
@@ -101,6 +101,9 @@ template <typename T> SetVector<T> getParentsOfType(Block &block);
 template <typename T>
 SetVector<T> getOperationsOfType(FunctionOpInterface funcOp);
 
+/// Return the condition used by \p branchOp.
+Value getCondition(RegionBranchOpInterface branchOp);
```

Comment:
Maybe. This function does the job and has the advantage that it can be extended to get the condition of for operations that do not implement that interface. We could pass an `Operation *` to the function and get it to work for any operation (in any dialect) that have a condition value (like LLVM::SelectOp which doesn't implement the RegionBranchOpInterface interface but has a condition).